client/protocols/analytics.py

```diff
@@ -8,9 +10,25 @@
 # TODO(albert): rename this InformationProtocol
 # Add all command line arguments here
 
+log = logging.getLogger(__name__)
+
 class AnalyticsProtocol(models.Protocol):
     """A Protocol that analyzes how much students are using the autograder."""
 
+    RE_SNIPPET = re.compile(r"""
+        \s*[\#\;]\s+BEGIN\s+(.*?)\n # \1 is question name
+        (.*)                        # \2 is the contents in between
+        \s*[\#\;]\s+END\s+\1\n
+        """, re.X | re.I | re.S)
+
+    RE_DEFAULT_CODE = re.compile(r"""
+    ^\"\*\*\*\sREPLACE\sTHIS\sLINE\s\*\*\*\"$
```

Comment:
I would take off the `$` at the end, because if a piece of default code looks like this:

```
# BEGIN "Question 0"
"*** REPLACE THIS LINE ***"
return 5
# END "Question 0"
```